src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -569,12 +571,23 @@ ov::SoPtr<ov::ICompiledModel> ov::npuw::CompiledModel::compile_submodel(const st
                                                                         const std::string& device) {
     auto plugin = get_npuw_plugin();
     auto core = plugin->get_core();
+
     // set exclusive_async_requests in case when model is split
     // NOTE(dm): Not sure if it is required for the NPUW plugin, but likely it is
     auto& device_config = m_meta_devices[device];
+
+    if (!m_cache_dir.empty() &&
+        ov::util::contains(core->get_property(device, ov::supported_properties), ov::device::capabilities) &&
+        ov::util::contains(core->get_property(device, ov::device::capabilities),
+                           ov::device::capability::EXPORT_IMPORT) &&
+        ov::util::contains(core->get_property(device, ov::internal::supported_properties),
+                           ov::internal::caching_properties)) {
+        LOG_INFO("NPUW will try to utilize CACHE_DIR for " << submodel->get_friendly_name() << " submodel.");
+        device_config.insert(ov::cache_dir(m_cache_dir));
+    }
```

Comment:
you don't need to do these capabilities checks, Runtime checks that for you.

Just pass down `CACHE_DIR` to the devices when you ask them to compile.